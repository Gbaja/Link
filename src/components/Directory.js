import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

import { fetchDirectory } from "../actions/get_requests";
import SearchFormContainer from "./SearchForm/SearchFormContainer";
//import Pagination from "../Shared/Pagination";

class Directory extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 0,
      numberOfPages: []
    };
  }
  componentDidMount() {
    this.props.fetchDirectory(1, this.props.match.params.type).then(data => {
      this.setPageNumbers(this.props.directory.count);
    });
  }
  setPageNumbers = number => {
    let total = Math.ceil(number / 1);
    let arr = [];
    while (total > 0) {
      arr.unshift(total--);
    }
    this.setState({ numberOfPages: arr });
  };

  showPage = pageNum => () => {
    this.props.fetchDirectory(pageNum, this.props.match.params.type);
  };

  renderDetails = data => {
    if (this.props.match.params.type === "mentor") {
      return (
        <div key={data.id}>
          <p>
            {" "}
            Name : {data.firstName} {data.lastName}
          </p>
          <p> Current Job : {data.currentRole}</p>
          <p> Current Industry: {data.currentIndustry}</p>
          <p> Location: {data.location}</p>
          <p>
            {" "}
            <Link to={`profile/${data.accountType}/${data.id}`}>
              {" "}
              More info
            </Link>{" "}
          </p>
        </div>
      );
    } else {
      return (
        <div key={data.id}>
          <p>
            {" "}
            Name : {data.firstName} {data.lastName}
          </p>
          <p> Current Job : {data.currentMotive}</p>
          <p> Current Industry: {data.mentorIndustry}</p>
          <p> Location: {data.location}</p>
          <p>
            {" "}
            <Link to={`profile/${data.accountType}/${data.id}`}>
              {" "}
              More info
            </Link>{" "}
          </p>
        </div>
      );
    }
  };
  render() {
    if (Object.getOwnPropertyNames(this.props.directory).length === 0) {
      console.log("NONE");
      return <div>Loading</div>;
    } else {
      this.props.directory.count;
      const mentorsDetails = this.props.directory.rows;
      return (
        <div>
          <h1>All {`${this.props.match.params.type}s`}</h1>
          <SearchFormContainer />
          <h3> People </h3>
          {mentorsDetails.map(this.renderDetails)}
          {/* <Pagination numberOfPages={this.state.numberOfPages} /> */}
          {this.state.numberOfPages.map(num => {
            return (
              <span key={num} onClick={this.showPage(num)}>
                {num}
              </span>
            );
          })}
        </div>
      );
    }
  }
}
const mapStateToProps = state => ({
  directory: state.directory
});

export default connect(mapStateToProps, { fetchDirectory })(Directory);
