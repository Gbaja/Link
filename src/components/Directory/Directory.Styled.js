import styled from "styled-components";

export const DirectoryContainerDiv = styled.div`
  display: flex;
  flex-direction: column;
  @media screen and (min-width: 900px) {
    flex-direction: row;
  }
`;

export const DirectoryDiv = styled.div`
  width: 80%;
  margin: 0 auto;
`;

export const ListContainerDiv = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  @media screen and (min-width: 900px) {
    flex-direction: row;
  }
`;

export const PageNumbersDiv = styled.div`
  margin: 0 auto;
  text-align: center;
`;

export const PersonDiv = styled.div`
  width: 80%;
  border: 1px solid #e3c800;
  margin: 1rem auto;
  text-align: center;
  @media screen and (min-width: 900px) {
    width: 27%;
    margin: 1rem;
  }
`;

export const PersonInfoP = styled.p`
  padding: 0.3rem;
`;
export const PersonImg = styled.img`
  height: 100px;
  width: 100px;
  margin-top: 20px;
`;
