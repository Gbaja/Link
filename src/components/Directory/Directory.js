import React from "react";

import {
  DirectoryDiv,
  ListContainerDiv,
  PageNumbersDiv,
  PersonDiv,
  PersonInfoP,
  PersonImg,
  PageNumSpan
} from "./Directory.Styled";
import { FormsSubmitButton, Links } from "../Shared/Shared.styled";

const Directory = props => {
  const data = props.details;
  if (data.length === 0) {
    return (
      <div>
        <p>
          There are currently no mentors/mentees from your university signed up
          to Young & Giving
        </p>
      </div>
    );
  }
  return (
    <DirectoryDiv>
      <ListContainerDiv>
        {data.map(each => {
          return (
            <PersonDiv key={each.id}>
              {each.imageUrl ? (
                <PersonImg src={each.imageUrl} />
              ) : (
                <div>
                  <i
                    style={{
                      width: "100px",
                      height: "100px",
                      marginTop: "20px"
                    }}
                    className="fas fa-user-alt"
                  />
                </div>
              )}
              <PersonInfoP>
                {each.firstName} {each.lastName}
              </PersonInfoP>
              <PersonInfoP>
                {" "}
                {each.jobTitle}, {each.company}
              </PersonInfoP>
              <PersonInfoP> {each.industry}</PersonInfoP>
              <PersonInfoP>{each.location}</PersonInfoP>
              <PersonInfoP>
                {each.degree}, {each.graduationYear}
              </PersonInfoP>
              <FormsSubmitButton>
                <Links to={`/profile/${each.accountType}/${each.id}`}>
                  View profile
                </Links>
              </FormsSubmitButton>
            </PersonDiv>
          );
        })}
      </ListContainerDiv>
      <PageNumbersDiv>
        <PageNumSpan>Pages: </PageNumSpan>
        {props.numberOfPages.map(num => {
          return (
            <PageNumSpan key={num} onClick={props.showPage(num)}>
              {num}
            </PageNumSpan>
          );
        })}
      </PageNumbersDiv>
    </DirectoryDiv>
  );
};

export default Directory;
