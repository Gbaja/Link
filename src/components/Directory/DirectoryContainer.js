import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

import { fetchDirectory } from "../../actions/get_requests";
import SearchFormContainer from "../SearchForm/SearchFormContainer";
import Directory from "./Directory";
import Header from "../Shared/Header";

class DirectoryContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 0,
      numberOfPages: [],
      filter: {}
    };
  }

  componentDidMount() {
    console.log("NAME: ", this.props.filter.name);
    this.props
      .fetchDirectory(
        1,
        this.props.match.params.type,
        this.props.auth.universityName,
        this.state.filter.name,
        this.state.filter.location,
        this.state.filter.industry
      )
      .then(data => {
        this.setPageNumbers(this.props.directory.count);
      });
  }

  setPageNumbers = number => {
    let total = Math.ceil(number / 1);
    let arr = [];
    while (total > 0) {
      arr.unshift(total--);
    }
    this.setState({ numberOfPages: arr });
  };

  showPage = pageNum => () => {
    console.log(this.state.filter.location);
    this.props.fetchDirectory(
      pageNum,
      this.props.match.params.type,
      this.props.auth.universityName,
      this.state.filter.name,
      this.state.filter.location,
      this.state.filter.industry
    );
  };

  setFilter = filter => {
    console.log("SET FILTER: ", filter);
    this.setState({ filter });
    this.props
      .fetchDirectory(
        1,
        this.props.match.params.type,
        this.props.auth.universityName,
        filter.name,
        filter.location,
        filter.industry
      )
      .then(data => {
        this.setPageNumbers(this.props.directory.count);
      });
  };

  render() {
    console.log("FILTERRRRR: ", this.props.filter);
    if (Object.getOwnPropertyNames(this.props.directory).length === 0) {
      console.log("NONE");
      return <div>Loading</div>;
    } else {
      console.log(this.props.directory.rows);
      const mentorsDetails = this.props.directory.rows;
      return (
        <div>
          <Header />
          {this.props.auth.accountType === "University" ? (
            <Link to="/university_dashboard">Back to dashboard</Link>
          ) : this.props.auth.accountType === "Mentor" ? (
            <Link to="/mentor/dashboard">Back to dashboard</Link>
          ) : this.props.auth.accountType === "Mentee" ? (
            <Link to="/mentee/dashboard">Back to dashboard</Link>
          ) : (
            ""
          )}
          <SearchFormContainer setFilter={this.setFilter} />
          <Directory
            details={mentorsDetails}
            showPage={this.showPage}
            numberOfPages={this.state.numberOfPages}
          />
        </div>
      );
    }
  }
}
const mapStateToProps = state => ({
  directory: state.directory,
  auth: state.auth,
  filter: state.filter
});

export default connect(mapStateToProps, { fetchDirectory })(DirectoryContainer);
