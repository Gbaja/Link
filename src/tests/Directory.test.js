import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { DirectoryContainer } from "../components/Directory/DirectoryContainer";
import { Directory } from "../components/Directory/Directory";

const directory = {
  count: 2,
  rows: []
};

const auth = {
  accountType: "Mentor"
};

const match = {
  params: {
    type: "mentor"
  }
};

const details = [
  {
    id: 1,
    accountType: "Mentor",
    firstName: "Ola",
    lastName: "Gbaja",
    email: "ola@example.com",
    location: "UK",
    gender: "Female",
    ethnicity: "Black",
    imageUrl: "",
    universityName: "Durham university",
    degree: "Economics",
    graduationYear: "2011",
    jobTitle: "Dev",
    company: "FAC",
    dateStarted: "2011",
    endDate: "",
    industry: "tech",
    biography: "",
    reason: "",
    cv: true,
    shadow: true,
    interview: false,
    job: false,
    postgrad: true,
    career: true,
    socialMediaUrl: "",
    availability: "",
    createdAt: ""
  }
];
const mockShowPage = jest.fn();
const numberOfPages = [];
const mockFetchDirectory = jest.fn();
mockFetchDirectory.mockReturnValue(Promise.resolve());
describe("Directory component", () => {
  it("starts with the correct initial state", () => {
    //shallow runs componentdidmount, disable it to stop this

    const wrapper = shallow(
      <DirectoryContainer
        directory={directory}
        auth={auth}
        match={match}
        fetchDirectory={mockFetchDirectory}
      />,
      {
        disableLifecycleMethods: true
      }
    );
    expect(wrapper.state()).toEqual({
      loading: false,
      currentPage: 0,
      numberOfPages: [],
      filter: {}
    });
  });
  it("should update loading state correctly", () => {
    const wrapper = shallow(
      <DirectoryContainer
        directory={directory}
        auth={auth}
        match={match}
        fetchDirectory={mockFetchDirectory}
      />,
      {
        disableLifecycleMethods: true
      }
    );
    expect(wrapper.state().loading).toBeFalsy();
    wrapper.instance().componentDidMount();
    expect(wrapper.state().loading).toBeTruthy();
    wrapper.instance().componentDidUpdate({ directory: {} });
    expect(wrapper.state().loading).toBeFalsy();
  });

  it("Directory renders as stated", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Directory
            details={details}
            showPage={mockShowPage}
            numberOfPages={numberOfPages}
          />
        </MemoryRouter>
      )
      .toJSON();
    console.log("TREE: ", tree);
    expect(tree).toMatchSnapshot();
  });
});
