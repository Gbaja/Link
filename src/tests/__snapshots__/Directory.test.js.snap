// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component Directory renders as stated 1`] = `
<div
  className="sc-gZMcBi ghpCTO"
>
  <div
    className="sc-gqjmRU ZcGBA"
  >
    <div
      className="sc-jTzLTM gcAHAn"
    >
      <div>
        <i
          className="fas fa-user-alt"
          style={
            Object {
              "height": "100px",
              "marginTop": "20px",
              "width": "100px",
            }
          }
        />
      </div>
      <p
        className="sc-fjdhpX frdGgd"
      >
        Ola
         
        Gbaja
      </p>
      <p
        className="sc-fjdhpX frdGgd"
      >
         
        Dev
        , 
        FAC
      </p>
      <p
        className="sc-fjdhpX frdGgd"
      >
         
        tech
      </p>
      <p
        className="sc-fjdhpX frdGgd"
      >
        UK
      </p>
      <p
        className="sc-fjdhpX frdGgd"
      >
        Economics
        , 
        2011
      </p>
      <button
        className="sc-EHOje kCmGdS"
      >
        <a
          className="sc-htoDjs cAGtRz"
          href="/profile/Mentor/1"
          onClick={[Function]}
        >
          View profile
        </a>
      </button>
    </div>
  </div>
  <div
    className="sc-VigVT Dcaet"
  >
    <span
      className="sc-cSHVUG ivSyoe"
    >
      Pages: 
    </span>
  </div>
</div>
`;
